File: Euler009.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    int t, n, b, c;
    cin >> t;
    while (t--)
    {
        cin >> n;
        int max = -1;
        for (int i = 1; i <= n / 2; i++)
        {
y
            c = (int)sqrt(i * i + b * b);
            if (c + i + b == n)
                if (max < (i * b * c))
                    max = i * b * c;
        }
        if (max == 0)
            max = -1;
        cout << max << endl;
    }
}

File: euler009.py
import math

for _ in range(int(input())):
    n = int(input())
    max_ = -1
    for i in range(1, n // 2 + 1):
        b = (i * i - (i - n) * (i - n)) / (2 * (i - n))
        c = int(math.sqrt(i * i + b * b))
        if i + b + c == n:
            if max_ < i * b * c:
                max_ = i * b * c
    if max_ == 0:
        max_ = -1
    print(int(max_))